test/e2e/apiextensions_compositions_test.go

```diff
@@ -127,6 +127,39 @@ func TestBasicCompositionNamespaced(t *testing.T) {
 	)
 }
 
+func TestLackOfRightsNamespaced(t *testing.T) {
+	manifests := "test/e2e/manifests/apiextensions/composition/lack-of-rights-namespaced"
+	environment.Test(t,
+		features.NewWithDescription(t.Name(), "Test that when attempting to compose a resource the controller has insufficient rights to manage, we get correct messaging.").
+			WithLabel(LabelArea, LabelAreaAPIExtensions).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(config.LabelTestSuite, config.TestSuiteDefault).
+			WithSetup("CreatePrerequisites", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "setup/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "setup/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "setup/definition.yaml", apiextensionsv1.WatchingComposite()),
+			)).
+			Assess("CreateXR", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "xr.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "xr.yaml"),
+			)).
+			Assess("XRHasStatusField", funcs.AllOf(
+				funcs.ResourcesHaveFieldValueWithin(5*time.Minute, manifests, "xr.yaml", "status.coolerField", "I'M COOLER!"),
+				// A blank error is as good as we can do at the moment. This validates we get into a reconciliation error, which is better than nothing.
```

Comment:
why is it that we can't test the content of this condition's error message further? i assume because the message will be different each test run, but I'm not sure how/why yet 😇

Reply:
we don't have fuzzy matching in the content of the message in the tests.